app/src/main/java/com/kickstarter/viewmodels/BackingFragmentViewModel.kt

```diff
@@ -171,13 +171,13 @@ interface BackingFragmentViewModel {
 
             val backedProject = this.projectDataInput
                     .map { it.project() }
-                    .filter { it.isBacking }
 
-            val backing = backedProject
-                    .switchMap { it.slug()?.let { slug -> this.apolloClient.getProjectBacking(slug) } }
+            val backing = this.projectDataInput
+                    .switchMap { getBackingInfo(it) }
+                    .distinctUntilChanged()
```

Comment:
Why do we need to check for uniqueness here?

Reply:
You can  hit the view pledge button, then the back arrow and hit again view pledge really quick. that's why I added the check for uniqueness but honestly to not repeat the network call uneccessarly. But if you have a better way for avoiding that second call let me know!

Reply:
Ok, perhaps I'm misunderstanding the code, what I'm reading is:

1. `.distinctUntilChanged()` is placed after the call would have happened so I don't think it would prevent repeating the network call in this case?
2. `getBackingInfo(it:)` appears to return the backing immediately if it's passed in so if we already have one it won't make another request - would we already have one in the scenario you're describing?